featuretools/tests/entityset_tests/test_entity.py

```diff
@@ -72,5 +72,26 @@ def test_parents(es):
 
 
 def test_update_data(es):
-    # TODO: write test for this method, since it has new functionality
-    pass
+    df = es['customers'].df.copy()
+    df['new'] = [1, 2, 3]
+
+    with pytest.raises(ValueError) as excinfo:
+        es['customers'].update_data(df.drop(columns=['cohort']))
+    assert 'Updated dataframe is missing new cohort column' in str(excinfo)
+
+    with pytest.raises(ValueError) as excinfo:
+        es['customers'].update_data(df)
+    assert 'Updated dataframe contains 13 columns, expecting 12' in str(excinfo)
+
+
+def test_add_variable(es):
```

Comment:
rather than add a test for add_variable, let's just remove the function. it's only used in one place internally and we can probably avoid needing it at all